.github/workflows/macos-universal.yml

```diff
@@ -0,0 +1,207 @@
+      - name: Upload artifacts (x86_64)
+        uses: actions/upload-artifact@v4
+        with:
+          name: macos-x86_64-artifacts
+          path: |
+            openssl-${{ env.OPENSSL_VER }}-x86_64.tgz
+            libssh-${{ env.LIBSSH_VER }}-x86_64.tgz
+          if-no-files-found: error
+
+  macos-merge-universal:
+    name: Merge → universal2
+    runs-on: macos-14
+    needs: [macos-arm64-build, macos-x86_64-build]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: macos-arm64-artifacts
+          path: arm64
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: macos-x86_64-artifacts
+          path: x86_64
+
+      - name: Unpack
+        run: |
+          mkdir -p merge
+          tar xzf arm64/openssl-*-arm64.tgz -C merge
+          tar xzf arm64/libssh-*-arm64.tgz   -C merge
+          tar xzf x86_64/openssl-*-x86_64.tgz -C merge
+          tar xzf x86_64/libssh-*-x86_64.tgz  -C merge
+          ls -R merge
+
+      - name: Lipo → universal2
+        run: |
+          set -euxo pipefail
+          mkdir -p universal/openssl/lib universal/openssl/include universal/libssh/lib universal/libssh/include
+
+          cp -R merge/openssl/include/* universal/openssl/include/
+          cp -R merge/libssh/include/*   universal/libssh/include/
+
+          lipo -create merge/openssl/lib/libssl.dylib   merge/x86_64/openssl/lib/libssl.dylib   -output universal/openssl/lib/libssl.dylib
+          lipo -create merge/openssl/lib/libcrypto.dylib merge/x86_64/openssl/lib/libcrypto.dylib -output universal/openssl/lib/libcrypto.dylib
+          lipo -create merge/libssh/lib/libssh.dylib     merge/x86_64/libssh/lib/libssh.dylib     -output universal/libssh/lib/libssh.dylib
+
+          # optional static libs if built:
+          if [ -f merge/openssl/lib/libssl.a ] && [ -f merge/x86_64/openssl/lib/libssl.a ]; then
+            lipo -create merge/openssl/lib/libssl.a merge/x86_64/openssl/lib/libssl.a -output universal/openssl/lib/libssl.a
+          fi
+          if [ -f merge/openssl/lib/libcrypto.a ] && [ -f merge/x86_64/openssl/lib/libcrypto.a ]; then
+            lipo -create merge/openssl/lib/libcrypto.a merge/x86_64/openssl/lib/libcrypto.a -output universal/openssl/lib/libcrypto.a
+          fi
+          if [ -f merge/libssh/lib/libssh.a ] && [ -f merge/x86_64/libssh/lib/libssh.a ]; then
+            lipo -create merge/libssh/lib/libssh.a merge/x86_64/libssh/lib/libssh.a -output universal/libssh/lib/libssh.a
+          fi
+
+          file universal/openssl/lib/libssl.dylib
+          lipo -info universal/openssl/lib/libssl.dylib
+          otool -L universal/libssh/lib/libssh.dylib
+
+      - name: Package universal artifacts
+        run: |
+          tar czf openssl-${{ env.OPENSSL_VER }}-universal2.tgz -C universal openssl
+          tar czf libssh-${{ env.LIBSSH_VER }}-universal2.tgz   -C universal libssh
```

Comment:
having non-verbose invocations in the same step will result in a log smashed together